iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relocatable_ptr.inl

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_RELOCATABLE_POINTER_RELOCATABLE_PTR_INL
+#define IOX_HOOFS_RELOCATABLE_POINTER_RELOCATABLE_PTR_INL
+
+#include "iceoryx_hoofs/internal/relocatable_pointer/relocatable_ptr.hpp"
+
+namespace iox
+{
+namespace rp
+{
+template <typename T>
+relocatable_ptr<T>::relocatable_ptr(T* ptr)
```

Comment:
Please add `inline` to every template implementation otherwise we will get double definition errors.

Reply:
Yes, I forgot that. I really dislike the boilerplate needed and some of it purely due to the use of those inl files (or more precisely the out of class definition).